interface/src/instruction.rs

```diff
@@ -0,0 +1,1697 @@
+
+    /// Change the entity authorized to execute nonce instructions on the account
+    ///
+    /// # Account references
+    ///   0. `[WRITE]` Nonce account
+    ///   1. `[SIGNER]` Nonce authority
+    ///
+    /// The `Pubkey` parameter identifies the entity to authorize
+    AuthorizeNonceAccount(Pubkey),
+
+    /// Allocate space in a (possibly new) account without funding
+    ///
+    /// # Account references
+    ///   0. `[WRITE, SIGNER]` New account
+    Allocate {
+        /// Number of bytes of memory to allocate
+        space: u64,
+    },
+
+    /// Allocate space for and assign an account at an address
+    ///    derived from a base public key and a seed
+    ///
+    /// # Account references
+    ///   0. `[WRITE]` Allocated account
+    ///   1. `[SIGNER]` Base account
+    AllocateWithSeed {
+        /// Base public key
+        base: Pubkey,
+
+        /// String of ASCII chars, no longer than `pubkey::MAX_SEED_LEN`
+        seed: String,
+
+        /// Number of bytes of memory to allocate
+        space: u64,
+
+        /// Owner program account
+        owner: Pubkey,
+    },
+
+    /// Assign account to a program based on a seed
+    ///
+    /// # Account references
+    ///   0. `[WRITE]` Assigned account
+    ///   1. `[SIGNER]` Base account
+    AssignWithSeed {
+        /// Base public key
+        base: Pubkey,
+
+        /// String of ASCII chars, no longer than `pubkey::MAX_SEED_LEN`
+        seed: String,
+
+        /// Owner program account
+        owner: Pubkey,
+    },
+
+    /// Transfer lamports from a derived address
+    ///
+    /// # Account references
+    ///   0. `[WRITE]` Funding account
+    ///   1. `[SIGNER]` Base for funding account
+    ///   2. `[WRITE]` Recipient account
+    TransferWithSeed {
+        /// Amount to transfer
+        lamports: u64,
+
+        /// Seed to use to derive the funding account address
+        from_seed: String,
+
+        /// Owner to use to derive the funding account address
+        from_owner: Pubkey,
+    },
+
+    /// One-time idempotent upgrade of legacy nonce versions in order to bump
+    /// them out of chain blockhash domain.
+    ///
+    /// # Account references
+    ///   0. `[WRITE]` Nonce account
+    UpgradeNonceAccount,
+}
+
+/// Create an account.
+///
+/// This function produces an [`Instruction`] which must be submitted in a
+/// [`Transaction`] or [invoked] to take effect, containing a serialized
+/// [`SystemInstruction::CreateAccount`].
+///
+/// [`Transaction`]: https://docs.rs/solana-sdk/latest/solana_sdk/transaction/struct.Transaction.html
+/// [invoked]: crate::program::invoke
+///
+/// Account creation typically involves three steps: [`allocate`] space,
+/// [`transfer`] lamports for rent, [`assign`] to its owning program. The
+/// [`create_account`] function does all three at once.
+///
+/// # Required signers
+///
+/// The `from_pubkey` and `to_pubkey` signers must sign the transaction.
+///
+/// # Examples
```

Comment:
I know it's tedious, but can you fixup these examples to avoid using solana_program, and to instead use the component crates? or at the very least, have the examples use this crate for the instructions?